engine/crates/federated-graph/src/lib.rs

```diff
@@ -16,15 +16,28 @@ pub use from_sdl::{from_sdl, DomainError};
 
 #[derive(serde::Serialize, serde::Deserialize, Clone)]
 pub enum FederatedGraph {
-    V1(FederatedGraphV1),
+    V1(v1::FederatedGraphV1),
+    V2(v2::FederatedGraphV2),
 }
 
 impl FederatedGraph {
-    pub fn to_sdl(&self) -> Result<String, std::fmt::Error> {
+    pub fn into_sdl(self) -> Result<String, std::fmt::Error> {
         render_sdl(self)
     }
 
+    #[deprecated(note = "Use into_sdl() instead")]
+    pub fn to_sdl(&self) -> Result<String, std::fmt::Error> {
+        self.clone().into_sdl()
+    }
+
     pub fn from_sdl(sdl: &str) -> Result<FederatedGraph, DomainError> {
         from_sdl(sdl)
     }
+
+    pub fn into_v2(self) -> FederatedGraphV2 {
```

Comment:
`into_latest()` ? could be worth doing the same as Graeme did in `engine-v2-config`

Reply:
Ah I didn't have that in mind. `into_latest()` makes more sense and is more consistent so I'll do that.

Reply:
done in next commit